src/app/shared/fabric8-ui-http.service.ts

```diff
@@ -14,9 +15,43 @@ import { AUTH_API_URL, HttpService , SSO_API_URL } from 'ngx-login-client';
 import { Observable } from 'rxjs';
 import * as uuidV4 from 'uuid/v4';
 
+type PendingRequests = {
+  [cacheKey: string]: Observable<Response>;
+};
+
+function isGetMethod(url: string | Request, options?: RequestOptionsArgs): boolean {
+  return (typeof url === 'string' && (options == null || options.method === RequestMethod.Get)) ||
+    (url as Request).method === RequestMethod.Get;
+}
+
+// Creates a cache key from headers, query params, and URL
+function createCacheKey(req: string | Request, options?: RequestOptionsArgs): string {
+  let key;
+  if (typeof req === 'string') {
+    key = req;
+    if (options) {
+      if (options.params) {
+        key += `_${options.params.toString()}`;
```

Comment:
What would you think of introducing a hashing utility for cases like this where we need to construct cache keys? We do something similar in the DeploymentsService (https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/deployments/services/deployments.service.ts#L320), and although there's nothing particularly *wrong* about constructing the key in this way, there's also no particular reason for the key to be so potentially large or to directly encode state. Anyway, just a thought - I'm fine with this function as-is.